table/tables/cache_test.go

```diff
@@ -389,3 +394,55 @@ func TestCacheTableBatchPointGet(t *testing.T) {
 	tk.MustQuery("select * from bp_cache_tmp1 where id in (1, 4)").Check(testkit.Rows("1 11 101"))
 	tk.MustQuery("select * from bp_cache_tmp1 where u in (11, 14)").Check(testkit.Rows("1 11 101"))
 }
+
+func TestRenewLease(t *testing.T) {
+	// Test RenewLeaseForRead
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	se := tk.Session()
+	table := &model.TableInfo{
+		ID:                   1,
+		Name:                 model.NewCIStr("t"),
+		Charset:              "utf8",
+		Collate:              "utf8_bin",
+		TableCacheStatusType: model.TableCacheStatusEnable,
+		PKIsHandle:           true,
+	}
+	ctx := context.Background()
+	tbl := tables.MockTableFromMeta(table)
+	cacheTable := tbl.(table2.CachedTable)
+	defer cacheTable.CloseRenewCh()
+	err := se.NewTxn(ctx)
+	require.NoError(t, err)
+	txn, err := se.Txn(true)
+	require.NoError(t, err)
+	data := cacheTable.TryReadFromCache(txn.StartTS())
+	require.Equal(t, data, nil)
+	for {
+		err := se.NewTxn(ctx)
+		require.NoError(t, err)
+		txn, err := se.Txn(true)
+		err = cacheTable.UpdateLockForRead(se.GetStore(), txn.StartTS())
+		if err == nil {
+			break
+		}
+	}
+	startTs, _ := cacheTable.MockGetDataLease()
```

Comment:
please add some comment ... read the code to figure out what's it testing for is  ... a bit difficult

Reply:
okay. this is draft pr. ... i will add it.